exonum/src/node/consensus.rs

```diff
@@ -159,26 +160,32 @@ impl NodeHandler {
         {
             let mut schema = Schema::new(&mut fork);
             for raw in block.transactions() {
-                if let Some(tx) = self.blockchain.tx_from_raw(raw) {
-                    let hash = tx.hash();
-                    if schema.transactions().contains(&hash) {
-                        error!(
-                            "Received block with already known transaction, block={:?}",
-                            block
-                        );
-                        return None;
-                    }
-                    profiler_span!("tx.verify()", {
-                        if !tx.verify() {
-                            error!("Incorrect transaction in block detected, block={:?}", block);
+                match self.blockchain.tx_from_raw(raw) {
```

Comment:
Nitpick / debatable: imo, the following structure would look nicer (less indentation):

```rust
let tx = match self.blockchain.tx_from_raw(raw) {
    Ok(tx) => tx,
    Err(e) => {
        error!("{}, block={:?}", e.description(), block);
        return None;
    }
};

let hash = tx.hash();
// ...
```